src/corehost/cli/args.cpp

```diff
@@ -89,7 +89,8 @@ bool parse_arguments(const int argc, const pal::char_t* argv[], arguments_t& arg
             {
                 trace::error(_X("Failed to locate deps file: %s"), args.deps_path.c_str());
                 return false;
-            }            
+            }      
+            args.app_dir = get_directory(args.deps_path);      
             args.app_argc = args.app_argc - 1;
```

Comment:
I would expect app_dir to be where the app DLL is published and not where deps file is overriden with an argument.

You can raise an error if app_dir resolved above is different from deps file's directory. What is the scenario this might be required?